### PR TITLE
Add possibility to set number of threads for parallel run

### PR DESCRIPTION
**Pull request by eugenenekh**
To give user control of number of threads to use.
Default to 4.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -77,6 +78,12 @@
                     <argLine>
                         -javaagent:"${settings.localRepository}/org/aspectj/aspectjweaver/${aspectj.version}/aspectjweaver-${aspectj.version}.jar"
                     </argLine>
+                    <properties>
+                        <property>
+                            <name>dataproviderthreadcount</name>
+                            <value>${thread.count}</value>
```
Comment (mialeska) [inline pom.xml]: in TestNG there is an option called `threadcount`, so the name `thread.count` for other option is confusing.
http://testng.org/doc/documentation-main.html#running-testng

Comment (eugenenekh) [reply pom.xml]: any suggestions on naming? `dataproviderthreadcount`? `threads`?

Comment (mialeska) [reply pom.xml]: are you sure that we should use dataproviderthreadcount not threadcount?
if so, we can say data.provider.thread.count

Comment (eugenenekh) [reply pom.xml]: pretty sure it is `dataproviderthreadcount`
https://cucumber.io/docs/guides/parallel-execution/#testng
> The default thread count of the dataprovider in parallel mode is 10. To change this the `dataproviderthreadcount` property needs to be added to the configuration section of the Surefire or Failsafe plugin in the POM.

Comment (mialeska) [reply pom.xml]: year, but as I see we parallelized by classes, not by methods. I'm not sure, is this correct when we use `dataproviderthreadcount` option, does it take any effect?

Comment (eugenenekh) [reply pom.xml]: is there a way to limit threads in current version?

Comment (mialeska) [reply pom.xml]: threadcount? :)

Comment (eugenenekh) [reply pom.xml]: nope
```
<properties>
   <property>
       <name>threadcount</name>
       <value>${thread.count}</value>
   </property>
</properties>
```
does not work in our case

**Review by mialeska — APPROVED** [review]

**Comment by eugenenekh** [comment]
@mialeska, renamed parameter to `data.provider.thread.count`

**Pull request merged**